libraries/botbuilder-dialogs/botbuilder/dialogs/skills/skill_dialog.py

```diff
@@ -0,0 +1,233 @@
+            TurnContext.apply_conversation_reference(
+                activity,
+                TurnContext.get_conversation_reference(context.activity),
+                is_incoming=True,
+            )
+            activity.channel_data = context.activity.channel_data
+            activity.additional_properties = context.activity.additional_properties
+
+            await self._send_to_skill(context, activity)
+
+        await super().end_dialog(context, instance, reason)
+
+    @staticmethod
+    def _validate_begin_dialog_args(options: object) -> BeginSkillDialogOptions:
+        if not options:
+            raise TypeError("options cannot be None.")
+
+        dialog_args = BeginSkillDialogOptions.from_object(options)
+
+        if not dialog_args:
+            raise TypeError(
+                "SkillDialog: options object not valid as BeginSkillDialogOptions."
+            )
+
+        if not dialog_args.activity:
+            raise TypeError(
+                "SkillDialog: activity object in options as BeginSkillDialogOptions cannot be None."
+            )
+
+        # Only accept Message or Event activities
+        if (
+            dialog_args.activity.type != ActivityTypes.message
+            and dialog_args.activity.type != ActivityTypes.event
+        ):
+            raise TypeError(
+                f"Only {ActivityTypes.message} and {ActivityTypes.event} activities are supported."
+                f" Received activity of type {dialog_args.activity.type}."
+            )
+
+        return dialog_args
+
+    async def _send_to_skill(
+        self, context: TurnContext, activity: Activity,
+    ) -> Activity:
+        # Create a conversationId to interact with the skill and send the activity
+        conversation_id_factory_options = SkillConversationIdFactoryOptions(
+            from_bot_oauth_scope=context.turn_state.get(BotAdapter.BOT_OAUTH_SCOPE_KEY),
+            from_bot_id=self.dialog_options.bot_id,
+            activity=activity,
+            bot_framework_skill=self.dialog_options.skill,
+        )
+
+        skill_conversation_id = await self.dialog_options.conversation_id_factory.create_skill_conversation_id(
+            conversation_id_factory_options
+        )
+
+        # Always save state before forwarding
+        # (the dialog stack won't get updated with the skillDialog and things won't work if you don't)
+        skill_info = self.dialog_options.skill
+        await self.dialog_options.conversation_state.save_changes(context, True)
+
+        response = await self.dialog_options.skill_client.post_activity(
+            self.dialog_options.bot_id,
+            skill_info.app_id,
+            skill_info.skill_endpoint,
+            self.dialog_options.skill_host_endpoint,
+            skill_conversation_id,
+            activity,
+        )
+
+        # Inspect the skill response status
+        if not 200 <= response.status <= 299:
```

Comment:
In all languages we have a statuscodes set of constants, so you can do StatusCodes.OK. Can we have that in python? Not a fan of writing numbers where we could have semantically more meaningful code.

Also, C# has a method on responses called `IsSuccessStatusCode` which internally checks between 200 and 299. On top of te constants you could do that, optionally. If we do this in at least 2 places, I'd introduce the method right away,

Reply:
i had similar feedback on the other review we have constants so we should use them - in fact we seem to be using two different constants for http status code. Having said that - this is obviously a code quality issue rather than functionality - if this is an issue across more of the code don't block on this - just open a R9 bug for global clean up - not a big deal.